File: src/_/types/app-config.ts
export type ConfigEntry = {
  key: string;
  value: string;
};

export type ConfigSection = {
  name: string;
  entries: ConfigEntry[];
};

export type AppConfigData = {
  sections: ConfigSection[];
  config_path: string;
};

export type SaveAppConfigRequest = {
  sections: ConfigSection[];
};

export type SpInfo = {
  name: string;
  file_name: string;
};

export type SpExecutionResult = {
  name: string;
  success: boolean;
  message: string;
};

export type SpExecutionSummary = {
  total: number;
  success_count: number;
  error_count: number;
  results: SpExecutionResult[];
};

File: src-tauri/src/models/app_config.rs
//! Các kiểu dữ liệu cho cấu hình ứng dụng (config.ini) và quản lý Store Procedure.

use serde::{Deserialize, Serialize};

/// Một key-value pair trong một section của config.ini.
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct ConfigEntry {
    pub key: String,
    pub value: String,
}

/// Một section (nhóm) trong config.ini.
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct ConfigSection {
    pub name: String,
    pub entries: Vec<ConfigEntry>,
}

/// Toàn bộ nội dung config.ini, chia theo section.
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct AppConfigData {
    pub sections: Vec<ConfigSection>,
    pub config_path: String,
}

/// Request từ frontend khi lưu config.ini.
#[derive(Debug, Deserialize)]
pub struct SaveAppConfigRequest {
    pub sections: Vec<ConfigSection>,
}

/// Kết quả thực thi một stored procedure.
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct SpExecutionResult {
    pub name: String,
    pub success: bool,
    pub message: String,
}

/// Danh sách stored procedure có sẵn.
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct SpInfo {
    pub name: String,
    pub file_name: String,
}

/// Kết quả thực thi toàn bộ stored procedure.
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct SpExecutionSummary {
    pub total: usize,
    pub success_count: usize,
    pub error_count: usize,
    pub results: Vec<SpExecutionResult>,
}
